### PR TITLE
add loadbalancer rake task

### LINKED ISSUE #218: Add Capistrano task to remove application from the load balancer
## Acceptance Criteria

- [ ] Running `cap --hosts=<hosthere> staging application:remove_from_nginx` will prevent a node from serving traffic on the load balancer
- [ ] Running `cap --hosts=<hosthere> staging application:serve_from_nginx` will allow a node to serve traffic

## Technical Implementation

Add a health monitor that looks at a file on disk and reports a failure if it exists, so that nginx will fail its health check and pull it out.

Example PR: https://github.com/pulibrary/pdc_describe/pull/1985/files

We're really hoping `https://github.com/lbeder/health-monitor-rails/pull/127` will get merged.

## First Step

Update the health-monitor-rails gem so it has the new check.

### PR DESCRIPTION
**Pull request by kayiwa**
add two tasks:
one that allow us to remove an endpoint from serving to the
loadbalancer
another that will put it back on the loadbalancer

closes #218

### CODE REVIEW

**Review by carolyncole — CHANGES_REQUESTED** [review]
Thank you @kayiwa.  I think you will need to do a `bundle update --conservative health-monitor-rails` for this code to work.

**Comment by maxkadel** [comment]
There's actually a breaking change for Rails 7.2 in the 12.4.1 release, so you have to pin to release 12.4.0. You need to change the line in Gemfile for health-monitor-rails to `gem 'health-monitor-rails', '12.4.0'`

File: config/initializers/health_monitor.rb [hunk 1]
```diff
@@ -13,6 +13,10 @@
 
     config.path = :health
 
+    config.file_absence.configure do |file_config|
+      file_config.filename = "public/remove-from-nginx"
+    end
+
     config.error_callback = proc do |e|
       Rails.logger.error "Health check failed with: #{e.message}"
       Honeybadger.notify(e)
```
Comment (maxkadel) [inline config/initializers/health_monitor.rb:22]: In the example we don't report to Honeybadger if it's this specific check, so it doesn't get hammered

```suggestion
      Honeybadger.notify(e) unless e.is_a?(HealthMonitor::Providers::FileAbsenceException)
```

File: config/deploy.rb [hunk 2]
```diff
@@ -48,4 +48,36 @@
...
+    on roles(:app) do
+      within release_path do
+        execute :touch, "public/remove-from-nginx"
+      end
+    end
+  end
+
+  # You can/ should apply this command to a subset of hosts
+  # cap --hosts=oawaiver-staging2.lib.princeton.edu staging application:serve_from_nginx
+  desc "Marks the server(s) to be removed from the loadbalancer"
```
Comment (maxkadel) [inline config/deploy.rb:71]:
```suggestion
  desc "Marks the server(s) to be added back to the loadbalancer"
```

**Review by maxkadel — CHANGES_REQUESTED** [review]
Thanks Francis! Couple small changes.

File: config/deploy.rb [hunk 3]
```diff
@@ -48,4 +48,36 @@
   end
 end
 
+#   # You can/ should apply this command to a subset of hosts
+# cap --hosts=oawaiver-staging2.lib.princeton.edu staging application:remove_from_nginx
+desc "Marks the server(s) to be removed from the loadbalancer"
```
Comment (maxkadel) [inline config/deploy.rb:53]: Needs the namespace
```suggestion
namespace :application do
  desc "Marks the server(s) to be removed from the loadbalancer"
```

**Review by maxkadel — APPROVED** [review]
Thanks @kayiwa!

**Review by tpendragon — APPROVED** [review]

**Pull request merged**